services/app-api/src/resolvers/contract/submitContract.ts

```diff
@@ -448,3 +485,309 @@ export function submitContract(
+): Promise<void | Error> {
+    const contractRevisionID = contractRevision.id
+    const contractDocuments = contractRevision.formData.contractDocuments
+
+    if (!contractDocuments || contractDocuments.length === 0) {
+        // No documents to zip
+        return
+    }
+
+    try {
+        // Create an S3 key (destination path) for the zip file. This is where
+        // we are storing it in the S3 bucket.
+        const s3DestinationKey = `zips/contracts/${contractRevisionID}/contract-documents.zip`
+
+        // Generate the zip file and upload it to S3
+        const zipResult = await generateDocumentZip(
+            contractDocuments,
+            s3DestinationKey
+        )
+
+        if (zipResult instanceof Error) {
+            // Return the error to the caller
+            logError('generateContractDocumentsZip', zipResult)
+            if (span) {
+                setErrorAttributesOnActiveSpan(
+                    'contract documents zip generation failed',
+                    span
+                )
+            }
+            return zipResult
+        }
+
+        // Store zip information in database
+        const createResult = await store.createDocumentZipPackage({
+            s3URL: zipResult.s3URL,
+            sha256: zipResult.sha256,
+            contractRevisionID,
+            documentType: 'CONTRACT_DOCUMENTS',
+        })
+
+        if (createResult instanceof Error) {
+            logError(
+                'generateContractDocumentsZip - database storage failed',
+                createResult
+            )
+            if (span) {
+                setErrorAttributesOnActiveSpan(
+                    'contract documents zip database storage failed',
+                    span
+                )
+            }
+            return createResult
+        }
+
+        console.info(
+            `Successfully generated zip for contract documents: ${zipResult.s3URL}`
+        )
+        return
+    } catch (error: unknown) {
+        const errorMessage =
+            error instanceof Error ? error.message : String(error)
+        const err = new Error(
+            `Unexpected error in generateContractDocumentsZip: ${errorMessage}`
+        )
+        logError('generateContractDocumentsZip', err)
+        if (span) {
+            setErrorAttributesOnActiveSpan(
+                'contract documents zip generation failed',
+                span
+            )
+        }
+        return err
+    }
+}
+
+export async function createContractZips(
+    store: Store,
+    submitContractResult: ContractType,
+    span?: Span
+): Promise<void | Error> {
+    const contractRevision =
+        submitContractResult.packageSubmissions[0]?.contractRevision
+
+    if (!contractRevision) {
+        return
+    }
+
+    // Only attempt to generate zip if there are actually documents to zip
+    if (
+        contractRevision.formData.contractDocuments &&
+        contractRevision.formData.contractDocuments.length > 0
```

Comment:
Should this be increased to > 1? I'm thinking if there's just 1 file, it doesn't need to be zipped either?

(same question for rates and supporting docs)

Reply:
Yeah that's a good point. I worry that we'd end up having cases where we expect zip files in places in s3, but end up with single doc files. Or we have data in the table for zips that are just regular files. Maybe the move is to turn off the "download zip" button in the UI if there is only one file in the contract/rate/supporting docs section.

I think the easiest move is to just always zip, but I can add an RSP topic for us to talk about it.